### DISCUSSION TITLE
Build and deploy DNNs on non-zynq and non- alveo FPGAs

### DISCUSSION THREAD

**Discussion by oaramoon** [Xilinx/finn#508] [Q&A] [answered]
Hi All,
I am trying to build and deploy the tfc_w1a1 network for an Artix-7 FPGA. Is there any good tutorial out there?

**Comment by nhma20** [comment, marked as answer]
This is the closest to a guide I have been able to find: https://github.com/jterry-x/finn-examples/tree/main/build/fpga_flow
It includes a testbench for the generated FINN IP which may be helpful to decipher how data is loaded into and out of the accelerator.
Hope that is useful

**Reply by oaramoon** [reply]
Thanks! that's what I was looking for!

**Reply by anguyen9630** [reply]
Just wondering if any of you guys know how to actually use the stitched IP to run the model on PyTorch or Torch... I am thinking of adding it to MicroBlaze but have no idea how to actually access it.......

**Reply by fsalmasri** [reply]
@oaramoon Did you manage to deploy your model on Artix-7 ?

**Comment by BrooksOlney** [comment]
It seems doing this for the TFC network isn't too difficult of a task - the provided code is pretty comprehensive and easy to adapt. However, I can't say the same about the CNV networks (e.g., the CIFAR-10 variant cnv-w1a1). Have you (or anyone) had any luck deploying those networks to an Artix-7 FPGA? I've been able to deploy it, but the accuracy is far lower than expected, like ~60%.

**Reply by jmduarte** [reply]
Here's a link to the code for converting CNV-W1A1 to deploy it on an Arty A7-100T FPGA: https://github.com/mlcommons/tiny_results_v0.7/tree/main/open/hls4ml-finn/code/ic/CNV-W1A1

This was done by @maltanar and others as part of the MLPerf Tiny v0.7 benchmark submission. Results are here: https://mlcommons.org/en/inference-tiny-07/ (click the Open tab)

**Reply by BrooksOlney** [reply]
That's the exact chip I'm working with, actually. This is so incredibly helpful, thank you so so much!

**Reply by anguyen9630** [reply]
Just wondering if any of you guys know how to actually use the stitched IP to run the model on PyTorch or Torch... I am thinking of adding it to MicroBlaze but have no idea how to actually access it.......
